aten/src/ATen/native/RNN.cpp

```diff
@@ -654,11 +558,30 @@ static c10::List<c10::intrusive_ptr<CellParamsBase>> gather_quantized_params_fp1
   TORCH_CHECK(params.size() % 4 == 0,
               "incorrect number of quantized RNN parameters FP16");
   for (size_t i = 0; i < params.size(); i += 4) {
+    c10::intrusive_ptr<LinearPackedParamsBase> packed_struct_ih =
+        cpp_custom_type_hack::cast<c10::intrusive_ptr<LinearPackedParamsBase>>(
+          static_cast<at::Tensor>(params[i]));
+    c10::intrusive_ptr<LinearPackedParamsBase> packed_struct_hh =
+        cpp_custom_type_hack::cast<c10::intrusive_ptr<LinearPackedParamsBase>>(
+          static_cast<at::Tensor>(params[i + 1]));
+
+    // NB: we install the bias from the gathered parameters here because
+    // in the "new world", the fp16 linear apply() method always expects
+    // the bias to be present in the packed struct. In the "old world",
+    // we called `fbgemm_linear_fp16_weight_fp32_activation`, which took
+    // the bias explicitly and ignored the bias in the packed struct. To
+    // reconcile serialized models that behavied in the old style, we
+    // put the bias into the appropriate packed structures here.
+    //
+    // Hopefully we can remove this in the future when we eliminate
+    // the old style altogether
+    packed_struct_ih->set_bias(params[i + 2]);
```

Comment:
Hm, is it thread-safe? because technically we can call RNN from multiple threads at the same time and it could race.

In practice though it should be fine as we're just setting the pointer to the same value (so all refcounts will check out)

Reply:
What should we do here? lock?